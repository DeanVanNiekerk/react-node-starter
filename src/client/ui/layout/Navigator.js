import React from 'react';
import { Link } from 'react-router-dom';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem
} from 'reactstrap';

class Navigator extends React.Component {

    constructor(props) {

        super(props);

        this.state = {
            isOpen: false
        };

        this.toggle = this.toggle.bind(this);
    }

    toggle() {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    render() {

        return (
            <Navbar className="mb-3" color="primary" dark expand="md">
                <NavbarBrand href="/">
                    React App
                </NavbarBrand>
                <NavbarToggler onClick={this.toggle} />
                <Collapse isOpen={this.state.isOpen} navbar>
                    <Nav className="mr-auto" navbar>
                        <NavItem>
                            <Link className="nav-link" to="/users">Users</Link>
                        </NavItem>
                    </Nav>
                </Collapse>
            </Navbar>
        );
    }
}

export default Navigator



