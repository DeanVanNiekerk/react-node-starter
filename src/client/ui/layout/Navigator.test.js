import React from 'react';
import Navigator from './Navigator';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom' 

describe('Navigator', () => {

    it('renders correctly', () => {

        const tree = renderer
            .create(<Router><Navigator /></Router>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

});