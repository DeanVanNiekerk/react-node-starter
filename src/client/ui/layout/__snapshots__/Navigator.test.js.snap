// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator renders correctly 1`] = `
<nav
  className="mb-3 navbar navbar-expand-md navbar-dark bg-primary"
>
  <a
    className="navbar-brand"
    href="/"
  >
    React App
  </a>
  <button
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="mr-auto navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/users"
          onClick={[Function]}
        >
          Users
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
